### PR TITLE
Use form to perform logout instead of JavaScript

### PR DESCRIPTION
**Pull request by dryan**
Addresses #254

Replace the JavaScript method of logging out users with a traditional form POSTing to /logout. This takes care of CSRF issues from #254 and clears the session cookie which seems to be the root problem of #228 

Let's have someone make the form button look like the other links in the menu before this gets merged (or make that a new issue).

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -1,23 +1,16 @@
-from flask import (
-    current_app,
-    flash,
-    redirect,
-    render_template,
-    request,
-    session,
-    url_for,
-)
+from flask import (current_app, flash, redirect, render_template, request,
```
Comment (iandees) [inline app/auth/views.py]: Autoformatting makes it hard to review the important changes 😢

**Comment by iandees** [comment]
Looks like there might be tests somewhere assuming there's a single form on the page?

**Comment by dryan** [comment]
I changed all the forms I could find to have an `id` attribute and then updated the tests to look for the IDs. Let's be sure to do this going forward instead of relying on DOM order.

**Comment by iandees** [comment]
Nah this is OK. I was waiting for someone to touch-up the button before merging. Or we can extract that to another ticket.

**Comment by kcaffall** [comment]
Created #607 so that the UI request isn't buried in this backend issue.

**Comment by jillh510** [comment]
What's wrong with the test?

**Pull request merged**